api.planx.uk/modules/analytics/metabase/collection/service.ts

```diff
@@ -0,0 +1,90 @@
+import { MetabaseError, createMetabaseClient } from "../shared/client.js";
+import type { NewCollectionParams } from "./types.js";
+import { toSnakeCase } from "../shared/utils.js";
+
+const client = createMetabaseClient();
+
+export async function authentication(): Promise<boolean> {
+  try {
+    const response = await client.get("/user/current");
+    return response.status === 200;
+  } catch (error) {
+    console.error("Error testing Metabase connection:", error);
+    return false;
+  }
+}
+
+export async function newCollection({
+  name,
+  description,
+  parent_id,
+}: NewCollectionParams): Promise<any> {
+  try {
+    // Check if collection exists
+    const existingCollectionId = await checkCollections(name);
+    if (existingCollectionId) {
+      console.log(
+        `Collection "${name}" already exists with ID: ${existingCollectionId}`,
+      );
+      return existingCollectionId;
+    }
+
+    // If no existing collection, create new one
+    const requestBody = toSnakeCase({
+      name,
+      description,
+      parent_id,
+    });
+
+    // Remove undefined properties
+    Object.keys(requestBody).forEach(
+      (key) => requestBody[key] === undefined && delete requestBody[key],
+    );
```

Comment:
Passing through the `validate()` middleware should handle this step already - no additional keys would be present.